# HG changeset patch
# User Jason Riedy <jason@acm.org>
# Date 1224769881 14400
#      Thu Oct 23 09:51:21 2008 -0400
# Node ID 37afdd1d1bf880453a51f3712cf2686b3d948dc8
# Parent  738c5d74fcfcd3dc5a35efba6f33e798d4913859
Use FT2_CFLAGS when checking for FTGL headers.

The flags need added to both CXXFLAGS and CPPFLAGS to pass
all of AC_CHECK_HEADERS's tests.

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2008-10-23  Jason Riedy  <jason@acm.org>
+
+	* configure.in: Take care to use FT2_CFLAGS when checking
+	for FTGL headers.
+
 2008-10-22  David Bateman  <dbateman@free.fr>
 
 	* examples/@polynomial: Move examples/polynomial here.
 	* examples/@polynomial/Makefile.in: Update for new location.
 	* examples/Makefile.in: Ditto.
 	* configure.in: Change locate of examples/@polynomial makefile
 	in AC_CONFIG_FILES.
 	
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -649,28 +649,31 @@ OCTAVE_OPENGL
 if test -n "$OPENGL_LIBS"; then
   AC_DEFINE(HAVE_OPENGL, 1, [Define if OpenGL is available])
 
 ## ftgl (needs freetype 2)
   AC_CHECK_FT2([9.0.3],[],
     [warn_freetype="FreeType library not found. Native renderer will not have on-screen text"])
   if test -z "$warn_freetype"; then
     AC_LANG_PUSH(C++)
+      save_LIBS="$LIBS"
+      save_CXXFLAGS="$CXXFLAGS"
+      save_CPPFLAGS="$CPPFLAGS"
+      LIBS="$LIBS $OPENGL_LIBS $FT2_LIBS"
+      CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
+      CPPFLAGS="$CPPFLAGS $FT2_CFLAGS"
       has_ftgl_h=yes
       AC_CHECK_HEADERS([FTGL/ftgl.h ftgl.h], break, [
 	AC_CHECK_HEADERS([FTGL/FTGL.h FTGL.h], [
           AC_DEFINE(HAVE_FTGL_UPPERCASE,1,[Define to 1 if you have FTGL.h or FTGL/FTGL.h])
 	  break
 	  ], has_ftgl_h=no)])
       if test "$has_ftgl_h" = yes; then
 	AC_MSG_CHECKING([for FTGLTextureFont in -lftgl])
-        save_LIBS="$LIBS"
-	save_CXXFLAGS="$CXXFLAGS"
-	LIBS="$LIBS $OPENGL_LIBS $FT2_LIBS -lftgl"
-	CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
+	LIBS="$LIBS -lftgl"
         AC_TRY_LINK([
 #ifdef HAVE_FTGL_FTGL_H
 #ifdef HAVE_FTGL_UPPERCASE
 #include <FTGL/FTGL.h>
 #include <FTGL/FTGLTextureFont.h>
 #else
 #include <FTGL/ftgl.h>
 #endif
@@ -680,24 +683,26 @@ if test -n "$OPENGL_LIBS"; then
 #include <FTGLTextureFont.h>
 #else
 #include <ftgl.h>
 #endif
 #endif], [
 FTGLTextureFont font("");], [
           OPENGL_LIBS="$OPENGL_LIBS $FT2_LIBS -lftgl"
 	  LIBS="$save_LIBS"
+	  CPPFLAGS="$save_CPPFLAGS"
           AC_DEFINE(HAVE_FTGL, 1, [Define to 1 if FTGL is present])
 	  AC_MSG_RESULT(yes)
 	  XTRA_CXXFLAGS="$XTRA_CXXFLAGS $FT2_CFLAGS"],[
 	  AC_MSG_RESULT(no)
           warn_ftgl="FTGL library not found. Native renderer will not have on-screen text"
         ])
 	LIBS="$save_LIBS"
 	CXXFLAGS="$save_CXXFLAGS"
+	CPPFLAGS="$save_CPPFLAGS"
       else
         warn_ftgl="FTGL headers not found. Native renderer will not have on-screen text"
       fi
     AC_LANG_POP(C++)
   fi
 fi
 
 GRAPHICS_LIBS=
